packages/database/src/migrations/20210817064143-AddPacmossiTotalNumberOfMosquitosByMonthStackedBarChart-modifies-data.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+import { insertObject, generateId, findSingleRecord, findSingleRecordBySql } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const CODE = 'PMOS_Total_Mosquitos_Per_Month';
+const COUNTRIES = ['FJ', 'KI', 'NR', 'NU', 'PG', 'SB', 'TV', 'TO', 'VU'];
+const getDashboardCode = country => `${country}_PacMOSSI_Vector_Surveillance`;
+
+const REPORT_CONFIG = {
+  fetch: {
+    dataElements: ['PMOS_Anopheles', 'PMOS_Aedes', 'PMOS_Culex', 'PMOS_Mansonia'],
+    aggregations: [
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'field_station', aggregationEntityType: 'requested' },
+      },
+    ],
+  },
+  transform: [
+    'keyValueByDataElementName',
+    {
+      transform: 'aggregate',
+      period: 'group',
+      organisationUnit: 'drop',
+      '...': 'sum',
+    },
+    {
+      transform: 'select',
+      "'Anopheles'":
+        'divide($row.PMOS_Anopheles, sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia]))',
+      "'Anopheles_metadata'":
+        '{ numerator: $row.PMOS_Anopheles, denominator: sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia])}',
+      "'Aedes'":
+        'divide($row.PMOS_Aedes, sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia]))',
+      "'Aedes_metadata'":
+        '{ numerator: $row.PMOS_Aedes, denominator: sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia])}',
+      "'Culex'":
+        'divide($row.PMOS_Culex, sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia]))',
+      "'Culex_metadata'":
+        '{ numerator: $row.PMOS_Culex, denominator: sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia])}',
+      "'Mansonia'":
+        'divide($row.PMOS_Mansonia, sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia]))',
+      "'Mansonia_metadata'":
+        '{ numerator: $row.PMOS_Mansonia, denominator: sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia])}',
```

Comment:
I didn't know we can specify a JSON object here! That's awesome!

Reply:
Yup, mathjs supports basic objects, which can be quite useful! I'm happy with using this like in this case as this format is how the front-end wants to consume the data, but generally I think it may be better if we don't over-use this feature given that handling objects can be a bit tricky at times...

One other cool thing it would let us do though is have named parameters in expression functions, eg. `divide({numerator: $row.value, denominator: $row.total })` Such a syntax could be very useful for making our functions more human readable/user friendly 👍